lib/connection/pool.js

```diff
@@ -542,6 +544,12 @@ function messageHandler(self) {
           return handleOperationCallback(self, workItem.cb, new MongoError(err));
         }
 
+        // Look for clusterTime, and update it if necessary
+        if (message.documents[0] && message.documents[0].hasOwnProperty('$clusterTime')) {
+          const $clusterTime = message.documents[0].$clusterTime;
```

Comment:
what is the rationale behind doing this as opposed to just directly assigning `self.topology.clusterTime = message.documents[0].$clusterTime`? I don't see $clusterTime used again

Reply:
Good catch. The logic for `resolveClusterTime` used to be here, so this was more relevant.